### ISSUE TITLE
Incorrect inferred return type for __call__

### ISSUE THREAD

**Issue by mthuurne** [PyCQA/pylint#2199]
### Steps to reproduce

Run pylint on the following code:
```python
class Base:
    def __call__(self):
        return self

class Sub(Base):
    def submethod(self):
        pass

obj = Sub()
obj().submethod()
```

### Current behavior

The following issue is reported:
```
E: 10, 0: Instance of 'Base' has no 'submethod' member (no-member)
```

### Expected behavior

No issue should be reported, as `obj` is an instance of `Sub`.

### pylint --version output

```
No config file found, using default configuration
pylint 1.9.1, 
astroid 1.6.4
Python 3.4.6 (default, Mar 22 2017, 12:26:13) [GCC]
```

### More info

Incorrect type inference seems to be the problem here: the `Base` class indeed doesn't have a `submethod` method, but the call is made on an instance of the `Sub` class instead.

The that the use of `__call__` is crucial here: if a method with a different name is used, the false positive goes away.

This issue is identical to the one @samwebster reported in https://github.com/PyCQA/pylint/issues/981#issuecomment-248309904, but the original bug reported in that issue was traced to `super()`, which does not occur in the code fragment above. So I think this a separate problem that has the same symptoms but a different cause from #981, which is why I opened a new issue for it.

**Comment by brycepg** [comment]
Thanks for the report. Fixed this in astroid master

**Comment by mthuurne** [comment]
Thanks for fixing it.
